nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/InvokeHTTP.java

```diff
@@ -516,6 +519,15 @@
             )
             .build();
 
+    public static final PropertyDescriptor PROP_ENABLE_COOKIE_REDIRECTS = new PropertyDescriptor.Builder()
+            .name("enable-cookie-redirects")
+            .description("If true, when an HTTP server responds to a request with a redirect, the provided cookie will be copied to the following request.")
+            .displayName("Enable Cookie Redirects")
+            .required(true)
+            .defaultValue("false")
+            .allowableValues("true", "false")
```

Comment:
The configured cookie policy applies to all types of responses, not just redirects. For that reason, adjusting the property name would be helpful. Rather than having a `Boolean` value, it would be helpful to define this property as a strategy, defaulting to `Accept None`.  This would provide the opportunity for additional strategies down the road. A new `CookieStrategy` enum could be defined with values of `DISABLED` and `ACCEPT_ALL`.

```suggestion
    public static final PropertyDescriptor PROP_COOKIE_STRATEGY = new PropertyDescriptor.Builder()
            .name("cookie-strategy")
            .description("Strategy for accepting and persisting HTTP cookies. Accepting cookies enables persistence across multiple requests.")
            .displayName("Cookie Strategy")
            .required(true)
            .defaultValue(CookieStrategy.DISABLED)
            .allowableValues(CookieStrategy.values())
```

Reply:
You make good points, thanks for your input @exceptionfactory! I went ahead and implemented the changes you suggested. Let me know if they look alright.